### PR TITLE
Clear input value on Command + Delete for NumberField/ColorField

### LINKED ISSUE #1877: Command + Delete doesn't delete NumberField input value
# 🐛 Bug Report 

Originally reported here: https://twitter.com/haroenv/status/1390345899033956357?s=21

## 🤔 Expected Behavior

Command +Delete should delete the contents of the NumberField

## 😯 Current Behavior

Pressing Command + Delete when inside the NumberField doesn't delete the contents of the field.

## 💁 Possible Solution

We are currently preventing default for this input type: https://github.com/adobe/react-spectrum/blob/main/packages/@react-aria/textfield/src/useFormattedTextField.ts#L93-L95. A case should be added for `e.inputType === deleteSoftLineBackward` so that we don't prevent default. Other cases covered in https://www.w3.org/TR/input-events-1/#interface-InputEvent-Attributes should also be investigated

## 🌍 Your Environment

| Software         | Version(s) |
| ---------------- | ---------- |
| react-spectrum   | 3.10
| Browser          |
| Operating System |

## 🧢 Your Company/Team

RSP

### PR DESCRIPTION
**Pull request by reidbarber**
Closes #1877 

## ✅ Pull Request Checklist:

- [x] Included link to corresponding [React Spectrum GitHub Issue](https://github.com/adobe/react-spectrum/issues).
- [ ] Added/updated unit tests and storybook for this change (for new code or code which already has tests).
- [x] Filled out test instructions.
- [x] Updated documentation (if it already exists for this component).
- [x] Looked at the Accessibility Practices for this feature - [Aria Practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## 📝 Test Instructions:

For components that use useFormattedTextField (i.e. NumberField and ColorField):

1. Type text into input
2. Press CMD + Delete
3. Verify input is empty and remains active

### CODE REVIEW

File: packages/@react-aria/textfield/src/useFormattedTextField.ts [hunk 1]
```diff
@@ -77,6 +77,9 @@ export function useFormattedTextField(props: AriaTextFieldProps, state: Formatte
             ? input.value.slice(0, input.selectionStart - 1) + input.value.slice(input.selectionStart)
             : input.value.slice(0, input.selectionStart) + input.value.slice(input.selectionEnd);
           break;
+        case 'deleteSoftLineBackward':
```
Comment (LFDanLu) [inline packages/@react-aria/textfield/src/useFormattedTextField.ts:80]: Looks like Safari treats CMD+Delete as `deleteHardLineBackward` so you'll need to add that here as well.

**Review by LFDanLu — CHANGES_REQUESTED** [review]
Mind adding a test here as well? https://github.com/adobe/react-spectrum/blob/main/packages/@react-spectrum/numberfield/test/NumberField.test.js#L1915

File: packages/@react-aria/textfield/src/useFormattedTextField.ts [hunk 2]
```diff
@@ -77,6 +77,12 @@ export function useFormattedTextField(props: AriaTextFieldProps, state: Formatte
...
+          nextValue = input.value.slice(input.selectionStart);
+          break;
+        case 'deleteHardLineBackward':
+          nextValue = input.value.slice(input.selectionStart);
+          break;
```
Comment (snowystinger) [inline packages/@react-aria/textfield/src/useFormattedTextField.ts:85]: We should be able to simplify this to
```suggestion
        case 'deleteSoftLineBackward':
        case 'deleteHardLineBackward':
          nextValue = input.value.slice(input.selectionStart);
          break;
```

**Review by snowystinger — COMMENTED** [review]
Thanks for adding these to our tests! Some manual testing notes:

I don't think this is working correctly in Safari? I can't see CMD + Delete doing anything.

I think it's also not intended to be able to delete partial currency symbols, see https://reactspectrum.blob.core.windows.net/reactspectrum/50233b6d2f4b110c70f7a2badd5b76e61b1d76bc/storybook/index.html?path=/story/numberfield--currency-switcher
Set the Currency Display to Code, place the cursor in between two of the currency symbol letters, press Delete, nothing happens. Press CMD + Delete, it deletes a portion of the code.

What does state.validate return? That's the check to see if we should block the event and I'm surprised it's not blocking it while your test indicates it is.
https://github.com/adobe/react-spectrum/pull/2005/files#diff-0d27e9828095cc31dbe37a372f0dfc33464dfa4c1dd426d7952a3a6e8df28e25R99

File: packages/@react-aria/textfield/src/useFormattedTextField.ts (hunk 2 above)
Comment (reidbarber) [reply packages/@react-aria/textfield/src/useFormattedTextField.ts:85]: Thanks for catching that.

And it's working for me in Safari. Also not having the partial currency issue you mentioned. Looking into it now though.

Comment (snowystinger) [reply packages/@react-aria/textfield/src/useFormattedTextField.ts:85]: o, it hasn't built... i see the problem, 1 second i'll fix it

**Review by snowystinger — APPROVED** [review]
Tested across Chrome/FF/Safari, lgtm

**Review by LFDanLu — APPROVED** [review]
LGTM

**Pull request merged**